lib/puppet/pops/types/types.rb

```diff
@@ -701,6 +701,23 @@ def each(&block)
     block_given? ? r.each(&block) : r
   end
 
+  def generalize
+    # General form of an Enum is a String
+    PStringType::DEFAULT
+  end
+
+  def normalize(guard = nil)
+    # General form of an Enum is a String
+    case @values.size
+    when 0
+      PStringType::DEFAULT
+    when 1
+      PStringType.new(@values[0])
+    else
+      self
+    end
+  end
```

Comment:
Hm, so if we infer String for "abc", it is `String[3,3]` (but remembering that it is "abc"). It seems like one normalization is that it is *only* `String[3,3]` - i.e not compatible with `Enum[abc]`. Now it looks like it will be normalized to the same thing as `Enum[abc]`, only encoded in the String type instead of in an enum.

This logic looks wrong to me.